newer glibc requires explicit sys/stat.h inclusion when you use things
from that header

--- xfsdump-3.0.4/invutil/cmenu.c
+++ xfsdump-3.0.4/invutil/cmenu.c
@@ -21,6 +21,7 @@
 
 #include <ncurses.h>
 #include <signal.h>
+#include <sys/stat.h>
 
 #include "types.h"
 #include "mlog.h"
--- xfsdump-3.0.4/invutil/invidx.c
+++ xfsdump-3.0.4/invutil/invidx.c
@@ -21,6 +21,7 @@
 
 #include <ncurses.h>
 #include <sys/mman.h>
+#include <sys/stat.h>
 
 #include "types.h"
 #include "mlog.h"
--- xfsdump-3.0.4/invutil/stobj.c
+++ xfsdump-3.0.4/invutil/stobj.c
@@ -20,6 +20,7 @@
 #include <xfs/jdm.h>
 #include <ncurses.h>
 #include <sys/mman.h>
+#include <sys/stat.h>
 
 #include "types.h"
 #include "mlog.h"
